apps/login/src/lib/zitadel.ts

```diff
@@ -52,17 +58,33 @@ export async function getBrandingSettings(organization?: string) {
     async () => {
       return await settingsService
         .getBrandingSettings({ ctx: makeReqCtx(organization) }, {})
-        .then((resp) => resp.settings);
+        .then((resp) =>
+          resp.settings
+            ? toJson(BrandingSettingsSchema, resp.settings)
+            : undefined,
+        );
     },
-    ["branding"],
-    { revalidate: 3600, tags: ["branding"] },
-  )();
+    ["brandingSettings", organization ?? "default"],
+    { revalidate: 3600, tags: ["brandingSettings"] },
+  )().then((resp) =>
+    resp ? fromJson(BrandingSettingsSchema, resp) : undefined,
+  );
 }
 
 export async function getLoginSettings(orgId?: string) {
-  return settingsService
-    .getLoginSettings({ ctx: makeReqCtx(orgId) }, {})
-    .then((resp) => resp.settings);
+  return unstable_cache(
+    async () => {
+      return await settingsService
+        .getLoginSettings({ ctx: makeReqCtx(orgId) }, {})
+        .then((resp) =>
+          resp.settings
+            ? toJson(LoginSettingsSchema, resp.settings)
+            : undefined,
+        );
+    },
+    ["loginSettings", orgId ?? "default"],
+    { revalidate: 3600, tags: ["loginSettings"] },
```

Comment:
Shouldn't we default to a very short cache or none at all in favor of DevX?

Reply:
I will introduce a global variable which can be overwritten as env variable 👍